src/google-maps/map-kml-layer/map-kml-layer.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {
+  Directive,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subject} from 'rxjs';
+import {map, take, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/google-map';
+import {MapEventManager} from '../map-event-manager';
+
+/**
+ * Angular component that renders a Google Maps KML Layer via the Google Maps JavaScript API.
+ *
+ * See developers.google.com/maps/documentation/javascript/reference/kml#KmlLayer
+ */
+@Directive({
+  selector: 'map-kml-layer',
+})
+export class MapKmlLayer implements OnInit, OnDestroy {
+  private _eventManager = new MapEventManager(this._ngZone);
+  private readonly _options = new BehaviorSubject<google.maps.KmlLayerOptions>({});
+  private readonly _url = new BehaviorSubject<string>('');
+
+  private readonly _destroyed = new Subject<void>();
+
+  /**
+   * The underlying google.maps.KmlLayer object.
+   *
+   * See developers.google.com/maps/documentation/javascript/reference/kml#KmlLayer
+   */
+  kmlLayer?: google.maps.KmlLayer;
+
+  @Input()
+  set options(options: google.maps.KmlLayerOptions) {
+    this._options.next(options || {});
+  }
+
+  @Input()
+  set url(url: string) {
+    this._url.next(url);
+  }
+
+  /**
+   * See developers.google.com/maps/documentation/javascript/reference/kml#KmlLayer.click
+   */
+  @Output()
+  kmlClick: Observable<google.maps.KmlMouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.KmlMouseEvent>('click');
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/kml
+   * #KmlLayer.defaultviewport_changed
+   */
+  @Output()
+  defaultviewportChanged: Observable<void> =
+      this._eventManager.getLazyEmitter<void>('defaultviewport_changed');
+
+  /**
+   * See developers.google.com/maps/documentation/javascript/reference/kml#KmlLayer.status_changed
+   */
+  @Output()
+  statusChanged: Observable<void> = this._eventManager.getLazyEmitter<void>('status_changed');
+
+  constructor(private readonly _map: GoogleMap, private _ngZone: NgZone) {}
+
+  ngOnInit() {
+    if (this._map._isBrowser) {
+      this._combineOptions().pipe(take(1)).subscribe(options => {
+        // Create the object outside the zone so its events don't trigger change detection.
+        // We'll bring it back in inside the `MapEventManager` only for the events that the
+        // user has subscribed to.
+        this._ngZone.runOutsideAngular(() => this.kmlLayer = new google.maps.KmlLayer(options));
+        this._assertInitialized();
```

Comment:
Should we change the assertion function in a similar way to #19178 to avoid the non-null casts?

Reply:
Good idea, done.